tests/10apidoc/13ui-auth.pl

```diff
@@ -21,11 +24,35 @@ sub wait_for_cas_request
    });
 }
 
+# Generate a ticket request from the client to the homeserver (and "validate"
+# it with the CAS server).
+sub make_ticket_request
+{
+   my ( $http, $homeserver_info, $session, $ticket, $response ) = @_;
+
+   # Note that we skip almost all of the CAS flow since it isn't important
+   # for this test. The user just needs to end up back at the homeserver
+   # with a valid ticket (and the original UI Auth session ID).
+   my $login_uri = $homeserver_info->client_location . "/_matrix/client/r0/login/cas/ticket?session=$session&ticket=$ticket";
+
+   Future->needs_all(
+      wait_for_cas_request(
+         "/cas/proxyValidate",
+         response => $response,
+      ),
+      $http->do_request_json(
+         method   => "GET",
+         full_uri => $login_uri,
+         max_redirects => 0, # don't follow the redirect
+      ),
+   );
+}
+
 test "Interactive authentication types include SSO",
-   requires => [ local_user_fixture( with_events => 0 ), $main::API_CLIENTS[0] ],
+   requires => [ local_user_fixture( with_events => 0 ) ],
```

Comment:
0 is the default for with_events.

```suggestion
   requires => [ local_user_fixture() ],
```

likewise elsewhere.